homeassistant/components/kostal_plenticore/select.py

```diff
@@ -0,0 +1,129 @@
+            is_on=select.is_on,
+            device_info=plenticore.device_info,
+            unique_id=f"{entry.entry_id}_{select.module_id}",
+        )
+        for select in SELECT_SETTINGS_DATA
+    )
+
+
+class PlenticoreDataSelect(CoordinatorEntity, SelectEntity, ABC):
+    """Representation of a Plenticore Switch."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        plenticore: Plenticore,
+        entry_id: str,
+        platform_name: str,
+        device_class: str | None,
+        module_id: str,
+        data_id: str,
+        name: str,
+        current_option: str | None,
+        options: list[str],
+        is_on: str,
+        device_info: DeviceInfo,
+        unique_id: str,
+    ) -> None:
+        """Create a new switch Entity for Plenticore process data."""
+        super().__init__(
+            coordinator=SelectDataUpdateCoordinator(
+                hass,
+                _LOGGER,
+                "Select Data",
+                timedelta(seconds=30),
+                plenticore,
+            )
+        )
+        self.plenticore = plenticore
+        self.entry_id = entry_id
+        self.platform_name = platform_name
+        self._attr_device_class = device_class
+        self.module_id = module_id
+        self.data_id = data_id
+        self._attr_options = options
+        self.all_options = options
+        self._attr_current_option = current_option
+        self._is_on = is_on
+        self._device_info = device_info
+        self._attr_name = name or DEVICE_DEFAULT_NAME
+        self._attr_unique_id = unique_id
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        is_available = (
+            super().available
+            and self.coordinator.data is not None
+            and self.module_id in self.coordinator.data
+            and self.data_id in self.coordinator.data[self.module_id]
+        )
+
+        if is_available:
+            self._attr_current_option = self.coordinator.data[self.module_id][
+                self.data_id
+            ]
+
+        return is_available
+
+    async def async_added_to_hass(self) -> None:
+        """Register this entity on the Update Coordinator."""
+        await super().async_added_to_hass()
+        self.coordinator.start_fetch_data(
+            self.module_id, self.data_id, self.all_options
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Unregister this entity from the Update Coordinator."""
+        self.coordinator.stop_fetch_data(self.module_id, self.data_id, self.all_options)
+        await super().async_will_remove_from_hass()
+
+    async def async_select_option(self, option: str) -> None:
+        """Update the current selected option."""
+        self._attr_current_option = option
+        for all_option in self._attr_options:
+            if all_option != "None":
+                await self.coordinator.async_write_data(
```

Comment:
Do we need to run these calls sequentially or can we gather the calls and run them concurrently?

Reply:
Yes we need, as the options in the select are 2 seperate entities in the Kostal Plenticore UI with the "disadvantage", that both options need to be disabled, before you can enable one.